pkg/querier/chunk_store_queryable.go

```diff
@@ -32,8 +32,8 @@ type chunkStoreQuerier struct {
 	mint, maxt        int64
 }
 
-func (q *chunkStoreQuerier) Select(_ *storage.SelectParams, matchers ...*labels.Matcher) (storage.SeriesSet, error) {
-	chunks, err := q.store.Get(q.ctx, model.Time(q.mint), model.Time(q.maxt), matchers...)
+func (q *chunkStoreQuerier) Select(sp *storage.SelectParams, matchers ...*labels.Matcher) (storage.SeriesSet, error) {
+	chunks, err := q.store.Get(q.ctx, model.Time(sp.Start), model.Time(sp.End), matchers...)
```

Comment:
should this check that `sp != nil` ?  Maybe it can't be today, but it seems safer to follow the general pattern.

Reply:
This is only call by querier.go (Select), which includes that check.  Calling this with nil should panic IMO.  The only "root" queriers are the ones in querier.go and unified_querier.go, where the nil check need to be.